### PR TITLE
Dev/rafd 373

### PR DESCRIPTION
**Pull request by rishab-himmatramka-guavus**
Changes made for OWASP Top 10. 
Modules Updated are Cdap-Common, Cdap-Security, Cdap-App-Fabric.

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java [hunk 1]
```diff
@@ -488,7 +490,8 @@ private ProgramOptions createProgramOptions(File programOptionsFile) throws IOEx
    * Reads the content of the given file and decode it as json.
    */
   private <U> U readJsonFile(File file, Class<U> type) throws IOException {
-    try (Reader reader = Files.newBufferedReader(file.toPath(), StandardCharsets.UTF_8)) {
+	  String newPath = file.getCanonicalPath();
```
Comment (prashant-gour-guavus) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java:493]: What is the reason for doing this

File: cdap-common/.classpath [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (prashant-gour-guavus) [inline cdap-common/.classpath]: please remove this, as it should not be checked-in

File: cdap-common/.project [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (prashant-gour-guavus) [inline cdap-common/.project]: please remove this, as it should not be checked-in

File: cdap-common/.settings/org.eclipse.jdt.core.prefs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+eclipse.preferences.version=1
```
Comment (prashant-gour-guavus) [inline cdap-common/.settings/org.eclipse.jdt.core.prefs]: remove this file as well

File: cdap-common/.settings/org.eclipse.jdt.core.prefs [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.8
```
Comment (prashant-gour-guavus) [inline cdap-common/.settings/org.eclipse.jdt.core.prefs]: Remove this file as well

File: cdap-common/src/main/java/co/cask/cdap/common/conf/Constants.java [hunk 6]
```diff
@@ -21,6 +21,7 @@
 import com.google.common.collect.ImmutableMap;
 import com.google.inject.BindingAnnotation;
 
+
 import java.lang.annotation.ElementType;
```
Comment (prashant-gour-guavus) [inline cdap-common/src/main/java/co/cask/cdap/common/conf/Constants.java:22]: Do not check in empty lines

File: cdap-common/src/main/java/co/cask/cdap/common/lang/DirectoryClassLoader.java [hunk 7]
```diff
@@ -70,7 +70,8 @@ public DirectoryClassLoader(File dir, @Nullable String extraClassPath, ClassLoad
     Manifest manifest = null;
     try {
       try (
-        InputStream input = new FileInputStream(new File(dir, JarFile.MANIFEST_NAME.replace('/', File.separatorChar)))
+        InputStream input = new FileInputStream(new File(dir.getCanonicalPath()
```
Comment (prashant-gour-guavus) [inline cdap-common/src/main/java/co/cask/cdap/common/lang/DirectoryClassLoader.java:73]: Was this for line length

File: cdap-common/src/main/java/co/cask/cdap/common/utils/DirUtils.java [hunk 8]
```diff
@@ -178,8 +178,10 @@ public boolean accept(File dir, String name) {
    * Returns list of files under the given directory that are accepted by the given filter.
    * An empty list will be returned if the given file is not a directory.
    */
+  @SuppressWarnings("squid:S2083")
   public static List<File> listFiles(File directory, FilenameFilter filenameFilter) {
-    return listOf(directory.listFiles(filenameFilter));
+
```
Comment (prashant-gour-guavus) [inline cdap-common/src/main/java/co/cask/cdap/common/utils/DirUtils.java]: Remove this empty line

File: cdap-security/src/main/java/co/cask/cdap/security/server/LDAPLoginModule.java [hunk 9]
```diff
@@ -50,24 +50,25 @@
...
+    	
         @Override
         public X509Certificate[] getAcceptedIssuers() {
           return null;
         }
       }};
 
       try {
-        SSLContext sc = SSLContext.getInstance("SSL");
+        SSLContext sc = SSLContext.getInstance("SSL");
```
Comment (prashant-gour-guavus) [inline cdap-security/src/main/java/co/cask/cdap/security/server/LDAPLoginModule.java]: Remove all new lines and spaces wherever necessary

File: cdap-common/src/main/java/co/cask/cdap/common/lang/DirectoryClassLoader.java (hunk 7 above)
Comment (rishab-himmatramka-guavus) [reply cdap-common/src/main/java/co/cask/cdap/common/lang/DirectoryClassLoader.java:73]: Yes sir, It was giving checkstyle errors. Line length exceeding 120 characters.

**Pull request closed without merging**